### PR TITLE
Update project to support Compose

### PR DESCRIPTION
**Pull request by sav621**
## 💸 TL;DR

Adds support for Compose.

## 📜 Details
[R4A-9627](https://reddit.atlassian.net/browse/R4A-9627)

Screenshot_20220818-150217

## 🧪 Testing Steps / Validation

Tested with both Compose Activity as launcher Activity, as well as the current Activity.

## ✅ Checks

- [ ] CI tests (if present) are passing
- [ ] Adheres to code style for repo
- [ ] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,6 +1,10 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
-    ext.kotlin_version = "1.5.32"
+    ext {
+        kotlin_version = '1.5.21'
+        compose_version = '1.0.1'
```
Comment (sav621) [inline build.gradle]: Not the latest version, but I was unable to get a more recent version working with any Kotlin version.

Comment (rciovati) [reply build.gradle]: Just curious, is there any reason not to use a more recent version of kotlin too? As a candidate I might find it struggling to not be able to use recent APIs.

File: app/src/main/java/com/reddit/rickandmortyapp/compose/MainComposeActivity.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Main Activity for Compose support
+ * If you'd like to use Compose, please update the Android Manifest to reference this Activity
+ */
+class MainComposeActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            Scaffold(
```
Comment (DSteve595) [inline app/src/main/java/com/reddit/rickandmortyapp/compose/MainComposeActivity.kt]: Theming is a pretty universally used part of Compose, I think we should include either a MaterialTheme or a customized version (like the project wizard generates)

File: build.gradle (hunk 1 above)
Comment (DSteve595) [reply build.gradle]: Does this help? https://developer.android.com/jetpack/androidx/releases/compose-kotlin#pre-release_kotlin_compatibility

**Review by DSteve595 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (DSteve595) [reply build.gradle]: BTW as of very recently, the Compose compiler version (which is what matters for Kotlin compatibility) is decoupled from the Compose library version

Comment (sav621) [reply build.gradle]: Can I just go ahead and update to the latest Kotlin & Compose compiler versions?

Comment (DSteve595) [reply build.gradle]: Yeah why not

File: build.gradle [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+        compose_version = '1.3.0'
     }
 
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:4.2.2" // Be careful about upgrading this.
+        classpath 'com.android.tools.build:gradle:7.0.4' // Be careful about upgrading this.
```
Comment (sav621) [inline build.gradle:17]: Just want to make sure we're okay with this - I couldn't get the app building without this upgrade

Comment (DSteve595) [reply build.gradle:17]: Yup this is old enough that it's fine IMO

File: app/build.gradle [hunk 4]
```diff
@@ -42,36 +48,44 @@ repositories {
...
-    implementation 'androidx.appcompat:appcompat:1.4.0'
-    implementation 'androidx.constraintlayout:constraintlayout:2.1.2'
+    // Coroutine scope
+    implementation "androidx.lifecycle:lifecycle-runtime-ktx:$androidx_lifecycle_version"
+    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:$androidx_lifecycle_version"
+    implementation "androidx.lifecycle:lifecycle-livedata-ktx:$androidx_lifecycle_version"
+
+    // Material Design
+    implementation "androidx.compose.material:material:1.2.1"
+    implementation 'androidx.compose.material3:material3:1.0.0-alpha11'
```
Comment (DSteve595) [inline app/build.gradle:80]: Beta 1 is out now

File: app/build.gradle [hunk 5]
```diff
@@ -77,6 +77,7 @@ dependencies {
 
     // Material Design
     implementation "androidx.compose.material:material:1.2.1"
+    implementation 'androidx.compose.material3:material3:1.0.0-alpha11'
```
Comment (sav621) [inline app/build.gradle]: @DSteve595 This is what was included with a new Compose project, but it's an alpha version. Thoughts?

File: build.gradle [hunk 6]
```diff
@@ -1,12 +1,20 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
-    ext.kotlin_version = "1.5.32"
+    ext {
+        kotlin_version = '1.7.10'
+        compose_version = '1.3.0'
+        compose_ui_version = '1.2.1'
```
Comment (DSteve595) [inline build.gradle]: The distinction should be "compiler" vs "libraries", `ui` is just one of the libraries

**Review by DSteve595 — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 5 above)
Comment (DSteve595) [reply app/build.gradle]: Beta 1 is out now

File: app/build.gradle [hunk 7]
```diff
@@ -3,7 +3,7 @@ apply plugin: 'kotlin-android'
 apply plugin: 'kotlin-kapt'
 
 android {
-    compileSdkVersion 32
+    compileSdkVersion 33
```
Comment (sav621) [inline app/build.gradle:6]: Had to update this to update Material3 to the Beta

**Review by DSteve595 — APPROVED** [review]

**Pull request merged**